webpack.base.config.js

```diff
@@ -1,33 +1,45 @@
 const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+const path = require('path');
+const ForkTsCheckerWebpackPlugin = require('fork-ts-checker-webpack-plugin');
 
 module.exports = (env, argv) => {
     return {
-        mode: "production",
+        mode: 'production',
         entry: {
-            "device-observer": ["./src/index.js"]
+            'index': './src/index.ts'
         },
         output: {
-            filename: "[name].min.js",
+            filename: './[name].min.js',
```

Comment:
the *.d.ts. file should have the same name as result js file (min postfix should be removed)